From 1f3615f3cce1f71144e7bb606f95c0f2e87cd3b4 Mon Sep 17 00:00:00 2001
From: Pekka Paalanen <pekka.paalanen@collabora.com>
Date: Mon, 27 Jan 2020 12:15:20 +0200
Subject: [PATCH] tests: test-runner needs wayland-client

weston-test-runner.h includes wayland-util.h, therefore it needs
wayland-client. A partial dependency with just compile_args might have been
enough as it does not seem to use functions from wayland-util.c, but safer this
way and no harm.

Fixes: https://lists.freedesktop.org/archives/wayland-devel/2020-January/041149.html

Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>
---
 tests/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/meson.build b/tests/meson.build
index b9646921..a0735ad4 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -11,10 +11,14 @@ env_modmap += 'weston-test-desktop-shell.so=@0@;'.format(plugin_test_shell_deskt
 lib_test_runner = static_library(
 	'test-runner',
 	'weston-test-runner.c',
+	dependencies: dep_wayland_client,
 	include_directories: common_inc,
 	install: false,
 )
-dep_test_runner = declare_dependency(link_with: lib_test_runner)
+dep_test_runner = declare_dependency(
+	dependencies: dep_wayland_client,
+	link_with: lib_test_runner
+)
 
 lib_test_client = static_library(
 	'test-client',
-- 
2.25.0

